### PR TITLE
feat: simplification of color scheme

### PR DESCRIPTION
**Pull request by ahalyha**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. Use imperative, present tense in your commit description ("change", not "changed" or "changes") without uppercases or period (.) at the end.

# Checklist
- [ ] The implementation has been manually tested and complies with Textkernel [browser support guidelines](https://textkernel.com/browser-support/)
- [ ] The implementation complies with [accessibility](CONTIRBUTING.md#accessibility) standards.
- [ ] The component has a [displayName](CONTRIBUTING.md#display-names) defined.
- [ ] The component comes with a detailed [PropTypes](CONTRIBUTING.md#component-props) (and defaultProps) definition.
- [ ] Component PropTypes are sufficiently described / documented.
- [ ] There is [a story](CONTRIBUTING.md#component-showcases) in Storybook.

### CODE REVIEW

File: src/themes/oneui/base-variables/_header_footer.scss [hunk 1]
```diff
@@ -4,7 +4,7 @@ $header-foreground-color: var(--color-foreground) !default;
 $header-height: 50px !default;
 
 // Footer variables
-$footer-background-color: var(--color-accent) !default;
+$footer-background-color: var(--ccolor-secondary) !default;
```
Comment (mukiienko) [inline src/themes/oneui/base-variables/_header_footer.scss:7]: typo `--ccolor-secondary`

Comment (ahalyha) [reply src/themes/oneui/base-variables/_header_footer.scss]: done

File: src/components/Checkbox/Checkbox.scss [hunk 2]
```diff
@@ -40,7 +40,7 @@
 
         &:hover {
             .Checkbox__box {
-                border-color: var(--color-brand);
+                border-color: var(--color-bracolor-primary);
```
[the remark this answers is not in the document]
Comment (ahalyha) [reply src/components/Checkbox/Checkbox.scss]: yes, addressed

File: src/themes/OneUITheme/baseVariables/colors.ts [hunk 3]
```diff
@@ -48,5 +45,10 @@ export const computedColorVariables = (baseVariables: CssVars): CssVars => {
             baseVariables['--color-background'],
             20
         ),
+        '--color-accent-25': mixTwoColors(
```
Comment (eszthoff) [inline src/themes/OneUITheme/baseVariables/colors.ts:48]: can you explain what is the purpose of it, and why it is done differently then the other shade calculations?

Comment (mukiienko) [reply src/themes/OneUITheme/baseVariables/colors.ts:48]: We still use `--color-accent` and `--color-accent-25` in the Gauge component, so the guys decided to keep these colors. Maybe it makes sense to create dedicated color for Gauge `--color-gauge` @ahalyha ?

Comment (eszthoff) [reply src/themes/OneUITheme/baseVariables/colors.ts:48]: I agree if it is only needed in the Gauge, it can be defined there, and named in a more specific way.
Please also double check the function: --color-accent-25 vs the 3rd parameter.

Comment (ahalyha) [reply src/themes/OneUITheme/baseVariables/colors.ts:48]: We need to decide about Gaudge component. Should it be themeable or not?
If it should be themeable, here https://github.com/textkernel/oneui/pull/819/files#diff-7c78d6eec49fa52a674350828c100d0b7ed967697022856f1bbd8b1c2e3d40a8L81 the color defined that way: color-accent-25. If do not, we can leave 2 specific colors colors

File: package.json [hunk 4]
```diff
@@ -62,6 +62,7 @@
         "@types/enzyme": "3.10.12",
         "ajv": "7.2.4",
         "ajv-formats": "1.6.1",
+        "caniuse-lite": "1.0.30001397",
```
Comment (mukiienko) [inline package.json:65]: Tell me please why we need `caniuse-lite`?

File: src/themes/OneUITheme/baseVariables/colors.ts (hunk 3 above)
Comment (mukiienko) [reply src/themes/OneUITheme/baseVariables/colors.ts:48]: @ahalyha, now it is themeable, right? so let's keep it themeable. If you are talking about generating the gradients then I think we can split gradients building logic for context and for additional colors:
```
           &__gradientStart {
            &--context_accent {
                stop-color: var(--color-gauge-light);
            }
        }

        &__gradientEnd {
            &--context_accent {
                stop-color: var(--color-gauge-dark);
            }
        }

    @each $context in ($contexts) {
        &__gradientStart {
            &--context_#{ $context } {
                stop-color: var(--color-#{$context});
            }
        }

        &__gradientEnd {
            &--context_#{ $context } {
                stop-color: var(--color-#{$context}-25);
            }
        }
    }
```

File: src/components/SelectButtonGroup/SelectButtonGroup/SelectButtonGroup.tsx [hunk 5]
```diff
@@ -17,7 +17,9 @@ export interface Props<V>
     /** should the component take up all the width available */
     isBlock?: boolean;
     /** Color context for selected buttons */
-    context?: Context | null;
+    context?: Context | 'primary' | null;
+    /** Use primary color for selected buttons */
+    isPrimary?: boolean;
```
Comment (eszthoff) [inline src/components/SelectButtonGroup/SelectButtonGroup/SelectButtonGroup.tsx:22]: Same as for button. I find it confusing to have both `context` and `isPrimary`

File: src/components/SelectComponents/Autosuggest/Autosuggest.scss [hunk 6]
```diff
@@ -40,7 +40,7 @@
         }
 
         &::placeholder {
-            color: var(--color-muted);
+            color: var(--color-light);
```
Comment (eszthoff) [inline src/components/SelectComponents/Autosuggest/Autosuggest.scss:43]: same as above `neutral` vs `light`

File: src/components/SelectComponents/ComboboxMulti/ComboboxMulti.scss [hunk 7]
```diff
@@ -35,7 +35,7 @@
         }
 
         &::placeholder {
-            color: var(--color-muted);
+            color: var(--color-light);
```
Comment (eszthoff) [inline src/components/SelectComponents/ComboboxMulti/ComboboxMulti.scss:38]: same as above (`neutral` vs `light`)

File: src/components/Tabs/TabItem/TabItem.scss [hunk 8]
```diff
@@ -36,12 +36,12 @@
 
     &--disabled {
         cursor: not-allowed;
-        color: var(--color-muted);
+        color: var(--color-light);
```
Comment (eszthoff) [inline src/components/Tabs/TabItem/TabItem.scss:39]: same as above (`neutral` vs `light`)

File: src/components/Tabs/TabItem/TabItem.scss [hunk 9]
```diff
@@ -36,12 +36,12 @@
...
 
         &:hover,
         &:focus {
             border-bottom: 4px solid transparent;
-            color: var(--color-muted);
+            color: var(--color-light);
```
Comment (eszthoff) [inline src/components/Tabs/TabItem/TabItem.scss:44]: same as above

File: src/components/Teaser/Teaser.tsx [hunk 10]
```diff
@@ -36,12 +36,17 @@ export const Teaser: React.FC<Props> = ({
     ...rest
 }) => {
     const subTitleElement = (
-        <Text inline title={subTitle} context={disabled ? 'muted' : 'accent'} {...elem('subTitle')}>
+        <Text
+            inline
+            title={subTitle}
+            context={disabled ? 'neutral' : 'background'}
```
Comment (eszthoff) [inline src/components/Teaser/Teaser.tsx:42]: I think this is a mistake. Shouldn't it change to `{disabled ? 'neutral' : 'default'}`?
2 resons:
1. `context: background` is not something we normally have
2. text with background color will basically be invisible

File: src/components/Text/Text.scss [hunk 11]
```diff
@@ -11,5 +11,5 @@
         }
     }
 
-    @include context-attribute-with-color('color', (muted));
+    @include context-attribute-with-color('color', (light, primary));
```
Comment (eszthoff) [inline src/components/Text/Text.scss:14]: can you explain the reason for this solution?

File: src/components/TextArea/TextArea.tsx [hunk 12]
```diff
@@ -1,11 +1,9 @@
 import React, { forwardRef } from 'react';
 import { bem } from '../../utils';
 import styles from './TextArea.scss';
-import { Context, Size } from '../../constants';
+import { Size } from '../../constants';
 
 export interface Props extends React.HTMLAttributes<HTMLElement> {
-    /** The textarea context (e.g. brand, primary, bad, good etc. - defaults to brand) */
-    context?: Context;
```
Comment (eszthoff) [inline src/components/TextArea/TextArea.tsx:8]: I'm not sure we can get away without context here. I'm thinking of a form that has some validation that has a textArea in it, and validation fails for some reason.

File: src/components/Themeroller/Themeroller.tsx [hunk 13]
```diff
@@ -150,7 +150,7 @@ export const Themeroller: React.FC<Props> = ({
                         onChange={handleNameChange}
                     />
                     {Boolean(themeResultStore.fileName) && (
-                        <Text {...elem('themeName')} inline context="muted">
+                        <Text {...elem('themeName')} inline context="neutral-60">
```
Comment (eszthoff) [inline src/components/Themeroller/Themeroller.tsx:153]: I don't think this is a valid context

File: src/themes/OneUITheme/baseVariables/typography.ts [hunk 14]
```diff
@@ -12,7 +12,7 @@ export const typographyVariables = {
     '--h6-font-size': '10px',
     '--font-family-primary':
         '-apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Droid Sans, Helvetica Neue, sans-serif',
-    '--font-family-secondary':
+    '--font-family-info':
```
Comment (eszthoff) [inline src/themes/OneUITheme/baseVariables/typography.ts:15]: what problem does this change solve?

File: src/themes/oneui/mixins/text.scss [hunk 15]
```diff
@@ -21,6 +21,6 @@ $heading-font-sizes: $h1-font-size, $h2-font-size, $h3-font-size, $h4-font-size;
 @mixin heading($n: 1) {
     font-family: $heading-font-family;
     font-size: nth($heading-font-sizes, $n);
-    color: $color-secondary;
+    color: $color-primary;
```
Comment (eszthoff) [inline src/themes/oneui/mixins/text.scss:24]: this looks like it should have been broken already. Is this function used?

File: src/themes/oneui/placeholders/_input.scss [hunk 16]
```diff
@@ -23,11 +23,11 @@
 
     &:focus,
     &:hover {
-        border-color: var(--color-brand);
+        border-color: var(--color-primary);
     }
 
     &::placeholder {
-        color: var(--color-muted);
+        color: var(--color-light);
```
Comment (eszthoff) [inline src/themes/oneui/placeholders/_input.scss:30]: same as above (`neutral` vs `light`)

File: stories/molecules/TwoPaneView.tsx [hunk 17]
```diff
@@ -12,7 +12,7 @@ export const _TwoPaneView = (args) => (
         <LeftPane
             style={{
                 minHeight: '600px',
-                backgroundColor: 'var(--color-info-25)',
+                backgroundColor: 'var(--color-primary-25)',
```
Comment (eszthoff) [inline stories/molecules/TwoPaneView.tsx:15]: Is this change needed?

**Review by eszthoff — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -184,6 +184,13 @@ Use imperative, present tense in your commit description ("change", not "changed
 
 **Breaking changes** should be explicitly marked (singular, with uppercase) in the message footer, e.g. `BREAKING CHANGE: <note>`. This will result in a new `MAJOR` package version.
 
+Example of breaking change commit message:
+```
+feat(LabelPicker): new component
+
+BREAKING CHANGE: Type checking for Checkbox moved to TS and might break certain implementations that were not as intended by the original API
+```
+
```
Comment (eszthoff) [inline CONTRIBUTING.md:193]: Nice improvement!

File: src/themes/oneui/mixins/colors.scss [hunk 19]
```diff
@@ -1,4 +1,4 @@
-$contexts: (neutral, brand, primary, accent, info, good, warning, bad);
+$contexts: (neutral, primary, info, success, warning, danger);
```
Comment (eszthoff) [inline src/themes/oneui/mixins/colors.scss:1]: This is different from the change in `src/constants/style-related.ts`. I think this one is better, but maybe we should discuss. This actually effects a lot of the changes in consequence, so perhaps we should get to a conclusion about this before doing a lot of refactoring.

**Review by eszthoff — CHANGES_REQUESTED** [review]

File: src/components/Themeroller/Themeroller.tsx (hunk 13 above)
Comment (ahalyha) [reply src/components/Themeroller/Themeroller.tsx:153]: yes, you are right, i fixed it

File: src/themes/oneui/mixins/text.scss (hunk 15 above)
Comment (ahalyha) [reply src/themes/oneui/mixins/text.scss:24]: looks like we dont use it, as other functions from this file...

File: stories/molecules/TwoPaneView.tsx (hunk 17 above)
Comment (ahalyha) [reply stories/molecules/TwoPaneView.tsx:15]: changed accidentally, i returned it

File: src/themes/oneui/placeholders/_input.scss (hunk 16 above)
Comment (ahalyha) [reply src/themes/oneui/placeholders/_input.scss:29]: we checked all changes about colors with Lidiia last week, so, it should be correct :)

File: src/components/Buttons/Button/Button.tsx [hunk 20]
```diff
@@ -11,8 +11,12 @@ export interface Props
     > {
     /** The label of the button */
     children: NotEmptySingleReactNode;
-    /** The button context (e.g. brand, primary, bad, good etc. - defaults to neutral) */
+    /** The button context (e.g. info, danger, success etc. - defaults to neutral) */
     context?: Context | 'link';
+    /** Should the button have style like primary element */
+    isPrimary?: boolean;
+    /** Should the button have style like neutral element */
+    isNeutral?: boolean;
```
[the remark this answers is not in the document]
Comment (ahalyha) [reply src/components/Buttons/Button/Button.tsx]: we decided to remove context at all for most elements, so it will be or isPrimary/isNeutral or context props

File: src/themes/oneui/mixins/colors.scss (hunk 19 above)
Comment (ahalyha) [reply src/themes/oneui/mixins/colors.scss]: we decided to leave 3 types of context: success, warning, danger
and add other context types only for test

**Comment by antipin** [comment]
Closing in favour of the parallel branch `next` https://github.com/textkernel/oneui/pull/878
